### PR TITLE
Waveform extractor std

### PR DESCRIPTION
**Pull request by Saksham20**
@alejoe91 @samuelgarcia 
creating method to extract waveform standard deviation for every time point. 
Just like templates which are either a mean or median across different spikes, this method finds the standard deviation across the spikes for every time point. 
I think the get_templates method will work for mean.

### CODE REVIEW

File: spikeinterface/core/waveform_extractor.py [hunk 1]
```diff
@@ -197,7 +199,7 @@ def get_waveforms(self, unit_id, with_index=False):
         Returns
         -------
         wfs: np.array
-            The returned waveform (num_spikes, num_channels, num_samples)
+            The returned waveform (num_spikes, num_samples, num_channels)
```
Comment (Saksham20) [inline spikeinterface/core/waveform_extractor.py:202]: This is an error in orientation. Even in tests its this changed dimension order.

**Comment by alejoe91** [comment]
Hi @Saksham20 

I think it's easier to just change the `get_template` method and add the option `std` on top of average and median.

**Comment by Saksham20** [comment]
> Hi @Saksham20
> 
> I think it's easier to just change the `get_template` method and add the option `std` on top of average and median.

I wonder of the word 'template' would be good for std?
I was thinking template to be a representative waveform

**Comment by alejoe91** [comment]
Template stands for "average waveform". So in my opinion it also makes sense to also be able to return the template std. We could also have it as an optional return (e.g. with an extra arg `return_std=True/False`).

@samuelgarcia what do you think about it?

**Comment by samuelgarcia** [comment]
No. I think I is really easy and intuitive to have
```python
template_mean = we.get_template('mean')
template_std = we.get_template('std')
```

**Comment by Saksham20** [comment]
> No. I think I is really easy and intuitive to have
> 
> ```python
> template_mean = we.get_template(unit_id, 'mean')
> template_std = we.get_template(unit_id, 'std')
> ```
> 
> We could also add some quartile stuff with the same method so.

okay, makes sense to put std also under templates. Will make those changes

**Comment by Saksham20** [comment]
@alejoe91 @samuelgarcia let me know what you think of the latest changes

File: spikeinterface/core/tests/test_waveform_extractor.py [hunk 2]
```diff
@@ -64,6 +64,15 @@ def test_WaveformExtractor():
...
     assert templates.shape == (5, 210, 2)
 
+    wf_std = we.get_template(0, mode='std')
+    assert wf_std.shape == (210, 2)
+    wfs_std = we.get_all_templates(mode='std')
+    assert wfs_std.shape == (5, 210, 2)
+
+    wf_qnt = we.get_template(0, mode='quantile', quantile_value = 0.9)
+    assert wf_qnt.shape == (210, 2)
+    wf_qnt = we.get_all_templates(mode='std')
```
Comment (alejoe91) [inline spikeinterface/core/tests/test_waveform_extractor.py:74]:
```suggestion
    wf_qnt = we.get_all_templates(mode='quantile')
```

??

File: spikeinterface/core/waveform_extractor.py [hunk 3]
```diff
@@ -228,14 +232,14 @@ def get_template(self, unit_id, mode='median'):
         unit_id: int
             Unit id to retrieve waveforms for
         mode: str
-            'mean' or 'median' (default)
+            'mean', 'median' (default), 'std'(standard deviation), 'quantile', quantile_value
```
Comment (alejoe91) [inline spikeinterface/core/waveform_extractor.py:235]: `quantile_value` should be an extra argument with its own type and description

**Review by alejoe91 — CHANGES_REQUESTED** [review]
Thanks @Saksham20 

Just had a couple of small comments, the rest looks good :)

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**